snapcraft/__init__.py

```diff
@@ -377,6 +377,20 @@ def find_library(name):
     import ctypes.util
     ctypes.util.find_library = find_library
 
+
+def _get_version():
+    if _os.environ.get('SNAP_NAME') == 'snapcraft':
+        return _os.environ["SNAP_VERSION"]
```

Comment:
This is a cool trick, to get it from the env var. I don't see any potential issues, and it simplifies everything, even the tests. +1 to this approach.